src/daft-logical-plan/src/optimization/rules/split_udfs.rs

```diff
@@ -13,10 +13,77 @@ use itertools::Itertools;
 
 use super::OptimizerRule;
 use crate::{
-    ops::{Project, UDFProject},
+    ops::{Filter, Project, UDFProject},
     LogicalPlan,
 };
 
+/// Simple optimizer rule that checks if filters contain a UDF and if so, pulls it out of the filter.
+/// Expectation is that this rule will run before SplitUDFs, so that the UDFs are split out of the filter expression.
+#[derive(Default, Debug)]
+pub struct SplitUDFsFromFilters {}
+
+impl SplitUDFsFromFilters {
+    pub fn new() -> Self {
+        Self {}
+    }
+
+    pub fn try_optimize_filter(
+        &self,
+        count: &mut usize,
+        filter: &Filter,
+        plan: &Arc<LogicalPlan>,
+    ) -> DaftResult<Transformed<Arc<LogicalPlan>>> {
+        if is_udf(&filter.predicate) {
```

Comment:
This only checks if the top level is a UDF. What if you have a within the expression tree but not at the root?